src/shogun/evaluation/MachineEvaluation.cpp

```diff
@@ -121,5 +165,20 @@ CMachine* CMachineEvaluation::get_machine() const
 
 EEvaluationDirection CMachineEvaluation::get_evaluation_direction()
 {
-	return m_evaluation_criterion->get_evaluation_direction();
+	REQUIRE(m_evaluation_criteria->get_array_size()==1,
+		"Multiple Metrics provided. Please use get_evaluation_directions()");
+	CEvaluation* temp_eval = (CEvaluation*) m_evaluation_criteria->get_element_safe(0);
+	return temp_eval->get_evaluation_direction();
+}
+
+CDynamicArray<EEvaluationDirection> CMachineEvaluation::get_evaluation_directions()
+{
+	CDynamicArray<EEvaluationDirection> list_evaluation_directions = *(new CDynamicArray<EEvaluationDirection>());
+	int32_t num_eval = m_evaluation_criteria->get_num_elements();
+	for (int32_t i = 0; i < num_eval;i++)
+	{
+		CEvaluation* temp_eval = (CEvaluation*) m_evaluation_criteria->get_element_safe(i);
```

Comment:
put a comment that this is type safe by construction and thats why you dont check